### PR TITLE
feat(preset-uno + preset-icons): extract and update regexes

### PR DESCRIPTION
**Pull request by chu121su12**
Compared to #182, this PR targets part of the code base that requires some logic update. To note:
- `encodeSvg()`: extract to array and use reduce.
- `time()`: changed the second regex (`/ms|s$/`) to match the first one (`/(s|ms)$/`)

### CODE REVIEW

File: packages/preset-icons/src/utils.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-    .replace(/%/g, '%25')
-    .replace(/#/g, '%23')
-    .replace(/{/g, '%7B')
-    .replace(/}/g, '%7D')
-    .replace(/</g, '%3C')
-    .replace(/>/g, '%3E')
+  return svgCharRe.reduce(
+    (str, [regexp, replacement]) => str.replace(regexp, replacement),
+    svg.replace('<svg', (~svg.indexOf('xmlns') ? '<svg' : '<svg xmlns="http://www.w3.org/2000/svg"'))
+  )
```
Comment (antfu) [inline packages/preset-icons/src/utils.ts:18]: Is this way faster? If not, I would prefer the old one.

File: packages/preset-uno/src/utils/handlers/handlers.ts [hunk 2]
```diff
@@ -67,13 +67,14 @@ export function cssvar(str: string) {
     return `var(--${str.slice(1)})`
 }
 
+const timeRe = /(s|ms)$/
```
Comment (antfu) [inline packages/preset-uno/src/utils/handlers/handlers.ts:70]:
```suggestion
const timeRe = /(ms|s)$/
```

**Review by antfu — CHANGES_REQUESTED** [review]

File: packages/preset-icons/src/utils.ts (hunk 1 above)
Comment (userquin) [reply packages/preset-icons/src/utils.ts:18]: avoid the replacement when the svg xml namespace present?

File: packages/preset-uno/src/rules/behaviors.ts [hunk 3]
```diff
@@ -49,10 +49,13 @@ export const outline: Rule[] = [
...
+const listStyleRe1 = /-inside|-outside/
+const listStyleRe2 = /inside|outside/
+
 export const listStyle: Rule[] = [
-  [new RegExp(`^list-((${listStyleProps.join('|')})(?:(-outside|-inside))?)$`), ([, value]) => {
-    const style = value.split(/-outside|-inside/)[0]
-    const position = /inside|outside/.exec(value) ?? []
+  [new RegExp(`^list-((${listStyleProps.join('|')})(?:(-inside|-outside))?)$`), ([, value]) => {
+    const style = value.split(listStyleRe1)[0]
+    const position = listStyleRe2.exec(value) ?? []
```
[the remark this answers is not in the document]
Comment (chu121su12) [reply packages/preset-uno/src/rules/behaviors.ts:58]: This part only swaps inside|outside to match the surrounding style.

File: packages/preset-uno/src/utils/handlers/handlers.ts (hunk 2 above)
Comment (chu121su12) [reply packages/preset-uno/src/utils/handlers/handlers.ts]: Updated

File: packages/preset-icons/src/utils.ts (hunk 1 above)
Comment (chu121su12) [reply packages/preset-icons/src/utils.ts]: Reverted since I couldn't find noticeable speed difference

**Comment by antfu** [comment]
I found it's hard to read after this refactoring, thanks for your contributions but I don't think we want it.

**Pull request closed without merging**